mkt/submit/forms.py

```diff
@@ -107,6 +107,7 @@ def get_devices(self, source=None):
             source = self._get_combined()
 
         platforms = {'firefoxos': mkt.DEVICE_GAIA,
```

Comment:
You can probably replace that by dict + return by 

`return map(mkt.DEVICE_LOOKUP.get, source)`